docs/docs/resources/pipeline-components/kafka-app.yaml

```diff
@@ -5,21 +5,37 @@
   # Pipeline prefix that will prefix every component name. If you wish to not
   # have any prefix you can specify an empty string.
   prefix: ${pipeline_name}-
-  # Topic(s) from which the component will read input
   from: # Must not be null
     topics: # read from topic
       ${pipeline_name}-input-topic:
-        type: input # required
-        # role: topic-role # only used if type is `extra` or `extra-pattern`
+        type: input # Implied when role is NOT specified, alternatives of `type: null` or `role: null` also work
```

Comment:
I don't quite understand the comment. I also think that while it works with type or role null, that is more a side-effect of pydantic or whatever and should not really be documented as a feature

Reply:
I intended to say with the comment that the following are equal:

```
some-topic:
	type:
```
```
some-topic:
	type: ~
```
```
some-topic:
	type: null
```
```
some-topic:
	role:
```
```
some-topic:
	role: ~
```
```
some-topic:
	role: null
```

It would work the same with any attribute of `some-topic`


I originally wanted to allow default definitions of topics like the example bellow, but the added complexity wasn't worth it:
```
# Equivalent to specifying `type: input`
some-topic:
```

Reply:
I still think all this null stuff is more confusing than helpful. So I'd rather skip documenting it and treat it as a side-effect

Reply:
Done